pkg/model/bootstrapscript.go

```diff
@@ -358,14 +359,22 @@ func (b *BootstrapScript) Run(c *fi.Context) error {
 			}
 			return string(content), nil
 		},
+
+		"GzipBase64": func(data string) (string, error) {
+			return gzipBase64(data)
+		},
+	}
+
+	context := map[string]interface{}{
+		"compressStructs": b.ig.Spec.CompressUserData,
```

Comment:
I see `NodeUpSource` and `NodeUpSourceHash` are implemented as functions should we use the same approach?